backend/corpora/gallica/gallica.py

```diff
@@ -0,0 +1,124 @@
+
+
+class Gallica(XMLCorpusDefinition):
+
+    languages = ["fr"]
+    data_url = "https://gallica.bnf.fr"
+    corpus_ark = ""
+
+    def sources(self, start: datetime, end: datetime):
+        # obtain list of ark numbers
+        response = requests.get(
+            f"{self.data_url}/services/Issues?ark=ark:/12148/{self.corpus_ark}/date"
+        )
+        year_soup = BeautifulSoup(response.content, "xml")
+        years = [
+            year.string
+            for year in year_soup.find_all("year")
+            if int(year.string) >= start.year and int(year.string) <= end.year
+        ]
+        for year in years:
+            response = requests.get(
+                f"{self.data_url}/services/Issues?ark=ark:/12148/{self.corpus_ark}/date&date={year}"
+            )
+            ark_soup = BeautifulSoup(response.content, "xml")
+            ark_numbers = [issue_tag["ark"] for issue_tag in ark_soup.find_all("issue")]
+
+            for ark in ark_numbers:
+                source_response = requests.get(
+                    f"{self.data_url}/services/OAIRecord?ark={ark}"
+                )
+                if source_response:
+                    content_response = requests.get(
+                        f"{self.data_url}/ark:/12148/{ark}.texteBrut"
+                    )
+                    parsed_content = BeautifulSoup(
+                        content_response.content, "lxml-html"
+                    )
+                    yield (
+                        source_response.content,
+                        {"content": parsed_content},
+                    )
+
+    def content(self):
+        return FieldDefinition(
+            name="content",
+            description="Content of publication",
+            es_mapping=main_content_mapping(
+                token_counts=True,
+                stopword_analysis=True,
+                stemming_analysis=True,
+                language=self.languages[0],
+            ),
+            extractor=Metadata("content", transform=get_content),
+        )
+
+    def date(self, min_date: datetime, max_date: datetime):
+        return FieldDefinition(
+            name="date",
+            display_name="Date",
+            description="The date of the publication.",
+            es_mapping=date_mapping(),
+            extractor=XML(
+                Tag("dc:date"),
+            ),
+            results_overview=True,
+            search_filter=DateFilter(
+                min_date, max_date, description="Search only within this time range."
+            ),
+            visualizations=["resultscount", "termfrequency"],
+            csv_core=True,
+        )
+
+    def identifier(self):
+        return FieldDefinition(
+            name="id",
+            display_name="Publication ID",
+            description="Identifier of the publication on Gallica",
+            es_mapping=keyword_mapping(),
+            extractor=XML(Tag("dc:identifier"), transform=lambda x: x.split("/")[-1]),
```

Comment:
I think this might result in an IndexError if it cannot split the contents of the `dc:identifier` element.